Softeq.XToolkit.WhiteLabel/Services/DefaultJsonSerializer.cs

```diff
@@ -0,0 +1,112 @@
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace Softeq.XToolkit.WhiteLabel.Services;
+
+/// <summary>
+///     A <see langword="class"/> implementing <see cref="IJsonSerializer"/> using the System.Text.Json APIs.
+/// </summary>
+public class DefaultJsonSerializer : IJsonSerializer
+{
+    private readonly JsonSerializerOptions? _options;
+
+    /// <summary>
+    ///     Initializes a new instance of the <see cref="DefaultJsonSerializer"/> class with the specified parameters.
+    /// </summary>
+    /// <param name="options">The serialization settings to use for the current instance.</param>
+    public DefaultJsonSerializer(JsonSerializerOptions? options = null)
+    {
+        _options = options ?? DefaultOptions;
+    }
+
+    /// <summary>
+    ///     Gets default serialization options.
+    /// </summary>
+    public static JsonSerializerOptions DefaultOptions { get; } = CreateDefaultOptions();
+
+    private static JsonSerializerOptions CreateDefaultOptions() => new()
+    {
+        Converters =
+        {
+            new DateTimeUtcJsonConverter()
+        },
+        DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+        NumberHandling = JsonNumberHandling.AllowReadingFromString,
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+        PropertyNameCaseInsensitive = false
+    };
+
+    /// <inheritdoc />
+    public string Serialize(object value)
+    {
+        return JsonSerializer.Serialize(value, _options);
+    }
+
+    /// <inheritdoc />
+    public TResult? Deserialize<TResult>(string? value)
+    {
+        if (value == null)
+        {
+            throw new ArgumentNullException(nameof(value));
+        }
+
+        if (value.Length == 0)
+        {
+            return default;
+        }
+
+        return JsonSerializer.Deserialize<TResult>(value, _options);
+    }
+
+    /// <inheritdoc />
+    public Task SerializeAsync(object value, Stream stream)
+    {
+        return JsonSerializer.SerializeAsync(stream, value, _options);
+    }
+
+    /// <inheritdoc />
+    public async Task<TResult?> DeserializeAsync<TResult>(Stream? stream)
+    {
+        if (stream == null)
+        {
+            throw new ArgumentNullException(nameof(stream));
+        }
+
+        if (stream.Length == 0)
+        {
+            return default;
+        }
+
+        return await JsonSerializer.DeserializeAsync<TResult>(stream, _options).AsTask().ConfigureAwait(false);
+    }
+}
+
+/// <summary>
+///     Custom System.Json converter that implements Newtonsoft.Json DateTime settings:
+///     (DateFormatHandling.IsoDateFormat, DateTimeZoneHandling.Utc).
+/// </summary>
+public sealed class DateTimeUtcJsonConverter : JsonConverter<DateTime>
+{
+    /// <inheritdoc />
+    public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        var dateTime = reader.GetDateTime();
+        return dateTime.ToUniversalTime();
```

Comment:
I think two lines was needed for tests? Maybe we can simplify this method like 
public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) =>
       return reader.GetDateTime().ToUniversalTime();

Reply:
I prefer to keep it as is because it's more comfortable for debugging. Optimization by code size will be done by the compiler :)